FusionIIIT/applications/hostel_management/models.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+from django.db import models
+from django.contrib.auth.models import User
+from applications.globals.models import ExtraInfo, Staff, Faculty
+from applications.academic_information.models import Student
+from applications.complaint_system.models import Caretaker
+from django.utils import timezone
+
+
+class HostelManagementConstants:
+    ROOM_STATUS = (
+        ('Booked', 'Booked'),
+        ('CheckedIn', 'Checked In'),
+        ('Available', 'Available'),
+        ('UnderMaintenance', 'Under Maintenance'),
+        )
+
+    DAYS_OF_WEEK = (
+            (0, 'Monday'),
+            (1, 'Tuesday'),
+            (2, 'Wednesday'),
+            (3, 'Thursday'),
+            (4, 'Friday'),
+            (5, 'Saturday'),
+            (6, 'Sunday')
+        )
+
+    BOOKING_STATUS = (
+    ("Confirmed" , 'Confirmed'),
+    ("Pending" , 'Pending'),
+    ("Rejected" , 'Rejected'),
+    ("Canceled" , 'Canceled'),
+    ("CancelRequested" , 'Cancel Requested'),
+    ("CheckedIn" , 'Checked In'),
+    ("Complete", 'Complete'),
+    ("Forward", 'Forward')
+    )
+
+
+
+class Hall(models.Model):
+    hall_id = models.CharField(max_length=10)
+    hall_name = models.CharField(max_length=50)
+    max_accomodation = models.IntegerField(default=0)
+    number_students = models.PositiveIntegerField(default=0)
+    hall_caretaker = models.ManyToManyField(Caretaker, on_delete=models.CASCADE)
```

Comment:
I just noticed that you are importing the Caretaker model from `complain_system` which we shouldn't do as the Caretaker in that module is very general, which is required for complaint_system whereas here we need only HallCaretakers. Also, some of the things in that model is hard-coded which may also cause problem in this module.

So, for Hall Caretakers, instead of having a `hall_caretaker` field in Hall model, create a new model with name `HallCaretaker` with the following fields:
* staff --> Foreign Key to Staff model
* hall --> Foreign Key to Hall model.

This change in database design is according to what was taught to us in DBMS course (there is a 1:n relationship between Caretaker and Hall, which should be resolved by Foreign Key in Caretaker model).